### PR TITLE
Issue 9: Getting started vignette

### LINKED ISSUE #9: Expand getting started vignette
The `epidist` [vignette](https://github.com/epinowcast/epidist/blob/main/vignettes/epidist.Rmd) is quite compact and possibly difficult to read for users not familiar with the package and methods already. I suggest expanding the writing to make it easier for new users to follow. The scope of this work would be to add further descriptions, links and references, figures, tables to the getting started vignette.

### PR DESCRIPTION
**Pull request by athowes**
## Description

This PR will close #9.
It is a draft of a rewritten version of the first "Get started" vignette.

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by seabbs** [comment]
This looks good so far. Give me a ping when you want a proper review

**Comment by athowes** [comment]
@seabbs @kgostic -- I'm not fully done with this but would be interested in some review / feedback to make sure that I'm staying on the right path.

If you'd prefer to wait until the vignette is more fully complete, then let me know.

**Comment by athowes** [comment]
I will now try fixing this `pkgdown` error so that you should hopefully be able to extract out the `.html` version of the vignette.

**Comment by athowes** [comment]
I don't know what is causing the issue with the `pkgdown` website on check. I can render it OK locally. Let me know if you have any idea. It is also there on the other PR so it must be something already in `main`.

**Comment by kgostic** [comment]
@seabbs -- can you help with this tomorrow?

File: vignettes/epidist.Rmd
Comment (kgostic) [inline vignettes/epidist.Rmd:56]:
```suggestion

```suggestion

If you want more technical details: the `epidist` package implements models following best practices described in [Park et al. 2023](https://www.medrxiv.org/content/10.1101/2024.01.12.24301247v1) and [Charniga et al. 2024](https://arxiv.org/abs/2405.08841).
```
```

File: vignettes/epidist.Rmd
Comment (kgostic) [inline vignettes/epidist.Rmd:59]: Here, I'd insert a few lines describing the required data schema, including the required fields, a definition for each field, and notes on data type requirements, e.g.:

```suggestion
Data should be formatted as a [`data.table`]() with the following columns for use within the `epidist` package:

* `case` (data type): a unique ID for each observed case
* `ptime` (data type): ...
* etc. 
```

File: vignettes/epidist.Rmd
Comment (kgostic) [inline vignettes/epidist.Rmd:112]: I think many non-statistical users see "at random" and think "unpredictably", but you're really trying to convey that this is a stochastic process. Consider rewording?

File: vignettes/epidist.Rmd
Comment (kgostic) [inline vignettes/epidist.Rmd:139]:
```suggestion
Here we suppose that the interval is daily, meaning that only the date of the primary or secondary event, not the exact time of each event, is reported.  (Figure \@ref(fig:cens)):
```

File: vignettes/epidist.Rmd
Comment (kgostic) [inline vignettes/epidist.Rmd:157]: Epidemiological delays are often ascertained via the secondary event (e.g. when estimating an incubation period the symptom onset event is much more likely to be observed and reported than the infection event).

File: vignettes/epidist.Rmd
Comment (kgostic) [inline vignettes/epidist.Rmd:205]: I think this could be nice, but raises questions about whether we should keep the "naive" model or just code it using brms in the example.

We should also choose a pretty short example delay interval if we want the adjusted estimate to be much better than the raw one.

**Review by kgostic — APPROVED** [review]
@athowes this looks great overall. Your writing is really clear and concise.

I do think there's a bit at the end that still needs to be fleshed out, but if you can finish off the end, I think this is a reasonable outline for a getting started vignette.

@seabbs may want to add a bit more about installation and contributing issues at the top.

I'm approving so as not to hold things up but it would be really nice to see a rendered version of the makrdown as I haven't yet seen the figures in situ.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:56]: Agree. Added with Bibtex. Arguably you might like the link?

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:112]: "occur with a delay drawn from the lognormal distribution" --> chance to reinforce the delay distribution

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:139]: Agree.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:205]: I think we should code it using `brms` in the example. Perhaps you could say it's the wrong place, but if someone is familiar with `y ~ x` synatx it might reinforce to them that our model is really just some fancy version of that.

Good point about choosing the simulation settings to make sure there is a difference. When I get to this I'll choose the simulation settings to demonstrate the value of using `epidist`.

**Comment by seabbs** [comment]
Can you rebase this so the `pkgdown` builds and then I will review (I thought about rebasing myself but it seemed a bit cheeky)

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:59]: I think this is a good suggestion. My only uncertainty is that I don't exactly know which columns are "hard requirements" for using the package and which are "there sometimes".

I suppose doing this will clarify this point for me (which is needed to do anyway).

**Comment by seabbs** [comment]
> gree on making sure people have the package installed, but would be tempted to have that displayed very prominently in the README and not in the "getting started" vignette. 

Agree

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:57]: I think either or works. Personally I find links more useful but I get the portability arg behind just giving a citation

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd]: Good point. I like the reword suggestions

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:165]: > **What happens if we had the symptom onset date for an individual but not the infection date? Discard their data? Then it's a non-representative sample?

Yes exactly (so no the methods here are not accounting for this) - `epinowcast` has a module for this (the missingness module) but its a tricky problem)

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:242]: > We should also choose a pretty short example delay interval

Or long distribution?

> I think we should code it using brms in the example.

I think I agree but we may look at this and change our mind if it seems a bit complicated

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:5]: Can you explain?

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:11]: ?

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:36]: Why cut?

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:32]: I quite like a summary - did you drop because you will do later or you don't think its a good idea?

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:93]: Note there may be some work spinning up on making a small package to carry these transforms around

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:129]: I would usually do this by going to long form data and then using the new variable as then you can have a legend. This is nice and clear as is though

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:154]:
```suggestion
(ref:cens) Interval censoring of the primary and secondary event times obscures the delay times. Note that while daily censoring is most common, both the primary and secondary events can have other delay intervals, and this scenario is supported by  `epidist`.
```

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:168]: yes good idea. We could reproduce one from one of the papers or role a new one. It feels like we could promote this to a issue vs doing here if we think it needs more thought.

In terms of what is in `epidist` I quite like the primary event bar plot with the dots for secondary events but can also see the clear case for it to be done on individual delays when explaining

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:177]: if we wanted to make a point about ascertainment here seems like it could be a good choice

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:196]: ruh roh. On the face of it I can't see a code issue. Can dig in if helpful

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:212]: Prove it to me with a plot

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:230]: There are other choices that provide more functionality but are heavier like `tidybayes` (heavier in the sense of them themselves having more dependencies).

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:233]: As we don't have this as a dep I think we would need to add it as a suggest which I think is a valid thing to do

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:236]:
```suggestion
Although some users may wish to work with fit directly, we will see in the following section that epidist provides functions to make common post-processing tasks easy. Additionally, users can leverage tools that support brms models for fit objects.
```

My suggestion seems a bit clunky but I more generally think we need to flag this info

Comment (seabbs) [reply vignettes/epidist.Rmd:236]: Also I see from this seciton why you went with manual extraction in the previous section. I like it from a teaching sense my only concern would be does it add more verbosity (answer don't know what do you think?)

File: vignettes/epidist.Rmd
Comment (seabbs) [inline vignettes/epidist.Rmd:241]: we could either show the simple model is wrong or we could show the truncated dist above more clearly vs the underlying dist? That might do a lot of what fitting a simple model would do?

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:242]: See above comment about maybe doing this via visualing the true dist and the truncated and censored dist. I think do that and then see if we need this? We might still as we can clearly show the summary parameters of the fits vs true (like the old readme but simpler).

**Review by seabbs — APPROVED** [review]
I really like this @athowes it is really clearly written and communicates the points very well. 

I put a few minor comments. My only other concern here is maybe length but uncertain.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:5]: I find that the `html_vignette2` format creates figures which are formatted strangely (low resolution, incorrect scale of different objects in the figure). I guess this is somewhat of a hack and there is a reason that we would want the vignette to be in `html_vignette2` format rather than `html_document2` format. If there is some way to make the figures less bad in `html_vignette2` please let me know.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:11]: This was so that "as described in Park et al. ([2024](http://127.0.0.1:36043/rmd_output/0/#ref-park2024estimating)) and Charniga et al. ([2024](http://127.0.0.1:36043/rmd_output/0/#ref-charniga2024best))." appears rather than e.g. [1], [2].

Ideally perhaps the `.bibtex` format would also include linking to the paper. I don't know if any of the `citation-style-language/` ones allow you to do that?

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:36]: I don't think this is crucial information for someone reading the "get started" to be presented with at the beginning. If someone cares about dependencies then likely they know that they can look at the `DESCRIPTION` file to figure out the dependencies.

The point about output being readily usable with other tools seems somewhat obvious to me. Are people writing packages where the output is only usable with `data.table` say?

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:129]: I would also usually do that but in some sense I was trying to make the number of lines used low.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:154]: Added.

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:5]: Okay fair enough. I don't have a strong preference here. I'm not clear if there is a reason for `html_vignette` apart from backwards compatible formatting and if its just that no issue not using

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:11]: Got you. Happy to drop the csl then

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:196]: I'm also confused about why this is wrong.

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:36]: > If someone cares about dependencies then likely they know that they can look at the DESCRIPTION file to figure out the dependencies.

Ah so this is because these aren't always going to be package dependencies. We might (and do here - looking at you `gt`) use packages that only appear in the suggests. This means that when people follow our install guides they won't always be able to run the vignette without either installing the suggests or manually knowing about them. 

> The point about output being readily usable with other tools seems somewhat obvious to me. Are people writing packages where the output is only usable with data.table say?

I think it might be obvious to a lot of people but I also think this could scare quite a large minority of people so at the cost of the single sentence seems worth keeping.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:212]: Agree and done.

Comment (athowes) [reply vignettes/epidist.Rmd:212]: Still IDK what's going on with the true density..

image

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:230]: I just didn't know an easier way to do this. Open to moving to `brms` / `posterior` / `tidybayes`. This is probably a one off thing for the vignette rather than a function in the package so not sure how much it matters.

Comment (athowes) [reply vignettes/epidist.Rmd:230]: We already have `posterior` in `Imports:`

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:233]: Yeah I think we will use `gt` again in other vignettes so I'll add it.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:236]: Tried:

In the following section, we show that `epidist` provides functions to make common post-processing tasks easy. Additionally, for users familiar with Stan and `brms`, they may like to work with `fit` directly. For example, any tool that supports `brms` fitted model objects will be compatible with `fit`.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:241]: Yeah having done the above I agree, and "fitting a model" to the obscured data isn't really doing much. It does do some things like teach people we are not doing anything really magic, just an altered `brms` model, but unclear if it's worth it.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:165]: Worth mentioning in this vignette that we don't deal with it then? Or leave that to something more advanced? I'd suggest leave to more advanced.

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd]: Hmm so the version we have here: https://github.com/parksw3/epidist-paper

definitely works as intended (we should talk a bit about why we are plotting from scratch here (is this just for now or is this a long term choice you have made).

Comment (seabbs) [reply vignettes/epidist.Rmd]: ah you aren't passing via an args list: https://github.com/parksw3/epidist-paper/blob/d34a461c4d7526438d509eec99025abda30687c7/R/plot.R#L68

do the parameters you are passing influence what is being plot?

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:165]: I think very briefly pointing it out (and at some later date linking to something more complex) might be a good user journey

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:246]: I think using poosterior might be a good idea. What you have right now is a bit scary

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:259]: yeeah agree - also for our impact factor we need to be magical ;)

**Comment by seabbs** [comment]
When ready for another review cna you reping using the interface at top right (as this gives a different form of notification that I filter for vs saying something like ready for review)

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:246]: Will try doing. I don't know the function in `posterior` to do this off the top of my head or if there is one.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd]: Plotting from scratch because it's easier for me to write, rather than it being a long term thought out choice or something like that.

And yes that fixed it, great spot, thanks!

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd]: Yep!

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:246]: I didn't find an easy way to do this with `posterior`.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:177]: Happy to leave to you and Katie to decide on this. I'd imagine putting ascertainment into a more complex vignette could be better (especially if there is any analysis the package can do to help you with it.).

**Comment by athowes** [comment]
Great, thanks all for the comments both.

I'm going to give this another hour or so try, and then I think we should clear it up and merge it in to get things moving. Happy to come back and iterate it again in future.

Have enjoyed writing this and the comments / feedback process by the way! Very smooth / logistically easy to handle.

File: vignettes/epidist.Rmd
Comment (seabbs) [reply vignettes/epidist.Rmd:179]: I think we should make a concise point here and link out to something more complex inn the future

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd]: I still think this is a good suggestion but would like to leave to a later issue as I am unsure of the input requirements we want, and also unsure of the columns we want.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:167]: I've tried something brief about this. Hard to know where exactly to put it.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd]: I think I've done this.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:36]: Added:

Finally, to run this vignette yourself, you will need the `data.table`, `purrr` and `ggplot2` packages installed.
Note that to work with outputs from `epidist` you do not need to use `data.table`: any tool of your preference is suitable.

File: vignettes/epidist.Rmd
Comment (athowes) [reply vignettes/epidist.Rmd:173]: Going to make an issue to address this.

**Review by seabbs — APPROVED** [review]
LGTM. This looks to have covered all review points from myself and @kgostic.

**Pull request merged**
